A29-xds-tls-security.md

```diff
@@ -0,0 +1,509 @@
+[`UpstreamTlsContext`][UTC] as described [here][CL-TS-comment].
+Note that we don't (currently) support [`transport_socket_matches`][CL-TS-matches].
+The security configuration extracted from the `UpstreamTlsContext` thus obtained
+is passed down to all the child policies and connections. How this is done is language dependent
+and is similar to how the implementations pass policy information down to child policies.
+For example, C-core uses channel args to pass security configuration down, whereas
+Java constructs a dynamic `SslContextProvider` that is directly usable by a sub-channel's
+TLS handshaker to build an `SslContext` for the pending TLS handshake.
+
+Only when a channel is using `XdsChannelCredentials`, gRPC uses the security configuration
+in `UpstreamTlsContext`. [`common_tls_context`][CTC] in the `UpstreamTlsContext` contains
+the required security configuration. See below for [`CommonTlsContext`][CTC-type] processing.
+
+A secure client at a minimum requires [`validation_context_certificate_provider_instance`][VCCPI1]
+inside [`combined_validation_context`][CVC] to be able to validate the server certificate
+in the TLS mode. If this field is not set, gRPC will NACK the CDS update. The client identity
+certificate is configured via the [`tls_certificate_certificate_provider_instance`][TCCPI]
+field which is optional. The client certificate is sent to the server if the server requests
+or requires it in the TLS handshake. If the server requires the client certificate but is not
+present then the TLS handshake fails.
+
+If any of the following fields are present, gRPC will NACK the CDS update:
+
+* [sni][SNI]
+* [allow_renegotiation][ALLOW-RENEG]
+* [max_session_keys][MAX-SESS-KEYS]
+
+[A27:CDS]: A27-xds-global-load-balancing.md#cds
+[CL-TS]: https://github.com/envoyproxy/envoy/blob/9aca65c395f01020080166e4795455addde167fa/api/envoy/config/cluster/v3/cluster.proto#L937
+[CL-TS-comment]: https://github.com/envoyproxy/envoy/blob/9aca65c395f01020080166e4795455addde167fa/api/envoy/config/cluster/v3/cluster.proto#L932
+[CL-TS-matches]: https://github.com/envoyproxy/envoy/blob/9aca65c395f01020080166e4795455addde167fa/api/envoy/config/cluster/v3/cluster.proto#L680
+[CTC]: https://github.com/envoyproxy/envoy/blob/7d4b2cae486b66b62ba0d3e1e348504699bea1bf/api/envoy/extensions/transport_sockets/tls/v3/tls.proto#L37
+[SNI]: https://github.com/envoyproxy/envoy/blob/7d4b2cae486b66b62ba0d3e1e348504699bea1bf/api/envoy/extensions/transport_sockets/tls/v3/tls.proto#L40
+[ALLOW-RENEG]: https://github.com/envoyproxy/envoy/blob/7d4b2cae486b66b62ba0d3e1e348504699bea1bf/api/envoy/extensions/transport_sockets/tls/v3/tls.proto#L47
+[MAX-SESS-KEYS]: https://github.com/envoyproxy/envoy/blob/7d4b2cae486b66b62ba0d3e1e348504699bea1bf/api/envoy/extensions/transport_sockets/tls/v3/tls.proto#L53
+
+##### Server Authorization aka Subject-Alt-Name Checks
+
+"Server authorization" is performed by the client where the client authorizes
```

Comment:
Why "Server authorization" is in quotes?

Reply:
Because I thought this is not a well-defined or understood term in the industry and instead of providing a long or technical definition here I wanted to convey the intuitive meaning of "server authorization".